### PR TITLE
Happy and sad paths for sign_up feature

### PR DESCRIPTION
**Pull request by danmalmx**
##PT Link
https://www.pivotaltracker.com/story/show/160582768

## Sign_up feature created

* Test (feature) for happy and sad path:
    * Name, email, password and password confirmed
    * Name taken, password blank, password not matching 

What I've learned - how to test for both the happy and sad feature path.

### CODE REVIEW

**Comment by danmalmx** [comment]
Completed

File: features/sign_up.feature [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        Then I fill in "Name" with "Daniel"
+        And I fill in "Email" with "dan@gmail.com"
+        And I fill in "Password" with "ThisIsMyPassword"
+        And I fill in "Password confirmation" with "ThisIsMyPassword"
+        And I click on "Create" link
+        Then I should see a message that states "Welcome! You have signed up successfully."
+    
+    Scenario: Sign up unsuccessfull - password too short
+        When I click on "Sign up" link
+        Then I fill in "Name" with "Daniel"
```
Comment (faraznaeem) [inline features/sign_up.feature]: `And`

File: features/sign_up.feature [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        And I fill in "Email" with "dan@gmail.com"
+        And I fill in "Password" with "This"
+        And I fill in "Password confirmation" with "This"
+        And I click on "Create" link
+        Then I should see a message that states "Password is too short"
+        
+    Scenario: Sign up unsuccessfull - email has been taken
+        When I click on "Sign up" link
+        Then I fill in "Name" with "Daniel"
```
Comment (faraznaeem) [inline features/sign_up.feature]: `And`

File: features/sign_up.feature [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        Then I should see a message that states "Name has already been taken"
+
+    Scenario: Sign up unsuccessfull - password can't be blank
+        When I click on "Sign up" link
+        Then I fill in "Name" with "Daniel"
+        And I fill in "Email" with "dan@gmail.com"
+        And I fill in "Password" with ""
+        And I fill in "Password confirmation" with ""
+        And I click on "Create" link
+        Then I should see a message that states "Password can't be blank"
```
Comment (faraznaeem) [inline features/sign_up.feature]: `And` in the other scenarios too.

**Review by faraznaeem — CHANGES_REQUESTED** [review]

**Review by faraznaeem — APPROVED** [review]

**Pull request merged**